source/event_stream_rpc_client.c

```diff
@@ -174,8 +178,12 @@ static void s_on_channel_shutdown_fn(
 
     if (connection->bootstrap_owned) {
         aws_mutex_lock(&connection->stream_lock);
-        aws_hash_table_clear(&connection->continuation_table);
+        aws_hash_table_foreach(&connection->continuation_table, s_mark_each_continuation_closed, NULL);
         aws_mutex_unlock(&connection->stream_lock);
+
+        /* Lock must NOT be held while invoking callbacks. */
+        aws_hash_table_foreach(&connection->continuation_table, s_complete_and_clear_each_continuation, NULL);
```

Comment:
If we don't swap this hash table out (the usual swap and drain we do with lists/arrays), is this in fact still thread safe? Is there any chance that this continuation_table gets touched at this point from another thread?

Reply:
If this is in fact safe, can you document why?

Reply:
yes